synapse/api/auth.py

```diff
@@ -633,10 +625,18 @@ def get_access_token_from_request(request):
             return query_params[0].decode("ascii")
 
     @defer.inlineCallbacks
-    def check_in_room_or_world_readable(self, room_id, user_id):
+    def check_user_in_room_or_world_readable(
+        self, room_id: str, user_id: str, allow_departed_users: bool = False
```

Comment:
this defaults to `False` for consistency with `check_user_in_room`, and for safety (if you forget to set it, it's safer if we deny access to users that should have it than grant access to users that should).